apis/engine/v1alpha1/workplan.go

```diff
@@ -42,7 +42,8 @@ type WorkplanSpec struct {
 	SecurityContext *corev1.PodSecurityContext `json:"securityContext,omitempty"`
 	// ServiceAccount with triggering-resource/configmaps/secrets watch/read permissions.
 	// Also used to run all associated pods
-	ServiceAccount string `json:"serviceAccount,omitempty"`
+	ServiceAccount string            `json:"serviceAccount,omitempty"`
+	NodeSelector   map[string]string `json:"nodeSelector,omitempty"`
```

Comment:
// Compute Resources required by the sidecar container.
	Resources core.ResourceRequirements `json:"resources,omitempty"`

	// If specified, the pod will be dispatched by specified scheduler.
	// If not specified, the pod will be dispatched by default scheduler.
	// +optional
	SchedulerName string `json:"schedulerName,omitempty"`

	// If specified, the pod's tolerations.
	// +optional
	Tolerations []core.Toleration `json:"tolerations,omitempty"`

Reply:
Also, need these fields. Should come from workflow.